apps/core/lib/core/snitch/data/model/line_item.ex

```diff
@@ -43,7 +43,7 @@ defmodule Core.Snitch.Data.Model.LineItem do
   #Money<:USD, 19.98000000>
   ```
   """
-  @spec update_price_and_totals([map()]) :: [map()]
+  @spec update_price_and_totals([map]) :: [map]
```

Comment:
The whole Logic looks good, ThumbsUp 👍  for using `Stream.map`.
However I have yet another approach, correct me if I am wrong.
Here we are calculating the `unit_selling_prices` in `update_price_and_totals`
it will return a `Enumerable`.
Which again we are passing to `set_price_and_total`, which takes exactly one `line_item` where on line no. `62`
we are using it(`unit_selling_prices`) to retrieve `unit_price` for one `variant_id`.

Could we have had a separate method which would calculate the `unit_price` for one `line_item` there and there itself. and we just filter out all the `nil` line_times
in the `update_price_and_totals`.
Something just like below:
For `update_price_and_totals`:
line_items |> Enum.map(set_price_and_total())

and calculate the `unit_price` inside another separate method which would be called from `set_price_and_total`

You see the responsibility of `update_price_and_totals` should be only to update the price and details and not calculate `unit_selling_prices`.

Any thoughts or alternate methods on this are appreciated. 😄 
cc: @ashish173  @pkrawat1

Reply:
@kartikjagdale Could you pls open an issue on line_items and price computation because this PR is about payments.

Reply:
I've documented why `update_price_and_totals` is like how it is:
>   ## Note
>  Selling prices of all `LineItem`s are fetched from the DB in a single query.